tests/test-components/components/wasi-http-v0.2.0-rc-2023-11-10/src/lib.rs

```diff
@@ -0,0 +1,95 @@
+wit_bindgen::generate!({
+    path: "wit",
+    world: "wasi:http/handler@0.2.0-rc-2023-11-10",
+});
+
+use helper::{ensure_eq, ensure_ok};
+use wasi::http::outgoing_handler;
+use wasi::http::types::{Headers, Method, OutgoingBody, OutgoingRequest, Scheme};
+use wasi::io::streams::StreamError;
+
+helper::define_component!(Component);
+
+impl Component {
+    fn main() -> Result<(), String> {
+        let url = url::Url::parse("http://localhost:8080").unwrap();
```

Comment:
It would probably be robust _enough_ for the test runner to bind a random port, unbind, then pass that port in to both the service and test.

There is also a hack for this which can _mostly_ resolve races where you bind a random port, open a connection to that port, then close the port without closing the connection which puts the socket into TIME_WAIT, then SO_REUSEADDR in the service.

Reply:
Yes, support for ephemeral ports is added in https://github.com/fermyon/spin/pull/2193, but I didn't want to have to reimplement it here. I will add that once #2193 is merged.